src/core/gui/toolbarMenubar/ColorToolItem.cpp

```diff
@@ -56,6 +78,54 @@ auto ColorToolItem::createItem(bool) -> GtkWidget* {
     return this->item.get();
 }
 
+void ColorToolItem::handleRelease(GtkGesture* gesture, int n_press, double x, double y, ColorToolItem* self) {
+    if (self->singleClickGesture.get() == gesture) {
+        if (n_press == 1) {
+            GtkWidget* colorWidget = gtk_event_controller_get_widget(GTK_EVENT_CONTROLLER(gesture));
+
+            // Set the colorWidget in the opacity toolbox before Control->toolColorChanged()
+            // is invoked when processing the action of the colorItem.
+            // Control->toolColorChanged() will then call opacityToolbox->update()
+            // in which will be used the colorWidget to show the opacity toolbox at
+            // its coordinates
+            self->opacityToolbox->setColorWidget(colorWidget);
+        }
+    }
+}
+
+bool ColorToolItem::handleEnter(GtkEventController* eventController, gdouble x, gdouble y, ColorToolItem* self) {
+    if (self->enterLeaveController.get() == eventController) {
```

Comment:
Since the eventController is already attached to the widget, this test is not necessary.

Not directly related to this, I have another question. According to the gtk document, the caller of the function takes ownership of the data, and is responsible for freeing it. Do I have to do something to free the memory allocated by the EventController in the destructor, or is this already automatically handled by GObjectSPtr?